helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ResourceMonitor.java

```diff
@@ -274,13 +275,22 @@ public void updateResourceState(ExternalView externalView, IdealState idealState
     Set<String> partitions = idealState.getPartitionSet();
     int replica;
     try {
-      replica = Integer.valueOf(idealState.getReplicas());
-    } catch (NumberFormatException e) {
-      _logger.info("Unspecified replica count for {}, skip updating the ResourceMonitor Mbean: {}", _resourceName,
-          idealState.getReplicas());
-      return;
+      String replicaCount = idealState.getReplicas();
+      if (!replicaCount.equals(ResourceConfig.ResourceConfigConstants.ANY_LIVEINSTANCE.name())) {
+        try {
+          replica = Integer.parseInt(replicaCount);
+        } catch (NumberFormatException ex) {
+          _logger.info(
+              "Unspecified replica count for {}, skip updating the ResourceMonitor Mbean: {}",
+              _resourceName, idealState.getReplicas());
+          return;
+        }
+      } else {
+        replica = -1;
+      }
```

Comment:
Can we improve the idealState.getNumReplicas()? Just make -1 when it is ANY_LIVEISNTANCE there.

Reply:
@junkaixue I've updated the code to use an existing IdealState method (getReplicaCount mentioned above by komal) that sets it to -1 when replica is ANY_LIVEINSTANCE. are you referring to the same ?